scripts/weekly-dep-report.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# this script is making an assumption that the repository contains at least two commits
+
+main_branch=${1:-"HEAD"}
+git checkout --quiet "${main_branch}"
+
+last_commit=$(git log --pretty=format:'%H' | head -n 1)
+first_commit_within_last_seven_days=$(git log --since=7.days --reverse --pretty=format:"%H" | head -n 1)
```

Comment:
```suggestion
last_commit=$(git rev-list HEAD -1)
first_commit_within_last_seven_days=$(git rev-list --reverse --since=7.days.ago HEAD | head -n 1)
```

This is a bit simpler since we don't need to invoke log